pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -4182,12 +4182,24 @@ func readRowsForTimeSeriesResult(rows driver.Rows, vars []interface{}, columnNam
 		groupBy, groupAttributes, metricPoint := readRow(vars, columnNames)
 		sort.Strings(groupBy)
 		key := strings.Join(groupBy, "")
+		keys = append(keys, key)
 		seriesToAttrs[key] = groupAttributes
 		seriesToPoints[key] = append(seriesToPoints[key], metricPoint)
 	}
 
+	// remove duplicates from keys array
+	encountered := map[string]bool{}
+	uniqKeys := []string{}
+
+	for _, element := range keys {
+		if encountered[element] == false {
+			encountered[element] = true
+			uniqKeys = append(uniqKeys, element)
+		}
+	}
```

Comment:
You could use the map from above `seriesToAttrs` for a unique set of keys in the `rows.Next()` loop itself.

Reply:
Good idea, updated.